ovn_k8s/common/kubernetes.py

```diff
@@ -17,22 +17,60 @@
 
 import ovs.vlog
 
+from os import path
 from ovn_k8s.common import exceptions
+from ovn_k8s.common.util import ovs_vsctl
 
+CA_CERTIFICATE = "/etc/openvswitch/k8s-ca.crt"
 vlog = ovs.vlog.Vlog("kubernetes")
 
 
+def _get_api_params():
+    ca_certificate = None
+    api_token = None
+    k8s_api_server = ovs_vsctl("--if-exists", "get", "Open_vSwitch", ".",
+                               "external_ids:k8s-api-server").strip('"')
+    if k8s_api_server.startswith("https://"):
+        if not path.isfile(CA_CERTIFICATE):
+            k8s_ca_crt = ovs_vsctl("--if-exists", "get", "Open_vSwitch",
+                                   ".", "external_ids:k8s-ca-certificate"
+                                   ).strip('"')
+            if k8s_ca_crt:
+                k8s_ca_crt = k8s_ca_crt.replace("\\n", "\n")
+                ca_file = open(CA_CERTIFICATE, 'w+')
+                ca_file.write(k8s_ca_crt)
+                ca_certificate = CA_CERTIFICATE
+        else:
+            ca_certificate = CA_CERTIFICATE
```

Comment:
Good point. I will add this bit of code:
```python
    if not variables.K8S_API_SERVER:
        k8s_api_server = ovs_vsctl("--if-exists", "get", "Open_vSwitch", ".",
                                   "external_ids:k8s-api-server").strip('"')
    else:
        k8s_api_server = variables.K8S_API_SERVER
```
Please correct me if I wrong, we are calling `_get_api_params` from two different scripts: watcher and ovn-k8s-cni-overlay. In case of watcher we already initialized all necessary variables and can use them. However, ovn-k8s-cni-overlay is being called from kubernetes as a cni driver, so it doesn't have these variables.